src/runtime/pyhandle.cs

```diff
@@ -0,0 +1,172 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Python.Runtime
+{
+    struct PyHandle : IDisposable
```

Comment:
Having regular structs `IDisposable` is a bad idea. The following two cases would cause the pointer to be copied and potentially lead to double free:

```csharp
var x = GetNewPyHandleToSomething();
var y = x;
x.Dispose();
y.Dispose();
```

```csharp
var x = GetNewPyHandleToSomething();
void DoSomethingWithHandle(PyHandle handle) {
  ...
  handle.Dispose();
}
DoSomethingWithHandle(x);
x.Dispose();
```

Reply:
This merely for easy using statement like `using(var list = PyList_New())`, people should know what they're doing when they using the C API, if they're not, everything on language sight are just smoke and mirrors. Just like you can always decref a wrong object.

Reply:
@amos402 but they only have to use raw C API if you expose it. In that sense `*Reference` types are fool-proof, unless you use `DagerousGetAddress`.

C-style API is a error-prone mental burden, which is totally unnecessary in this case.

Reply:
Not agree with it, since you can't control the user how write the code, these fool-proof are helpless.

Reply:
@amos402 users of Python.NET do not need to work with pointers at all.

In PRs for Python.NET I can safely skip attempts to verify correct refcount balancing if the code does not use `.DangerousGetAddress()`, which will make them easier to read.